brjs-core/src/main/java/org/bladerunnerjs/model/engine/AbstractRootNode.java

```diff
@@ -89,7 +89,7 @@ public void registerNode(Node node) throws NodeAlreadyRegisteredException
 	public void clearRegisteredNode(Node node) {
 		String normalizedPath = node.dir().getAbsolutePath();
 		List<Node> nodesForPath = nodeCache.get(normalizedPath);
-		nodesForPath.remove(node.getTypeName());
+		nodesForPath.remove(node);
```

Comment:
There's no point applying a fix to a piece of code that isn't even used :-) Can we try removing it? `nodeCache.get(..)` could return a copy of the array it holds, in which case that method would have no effect on the real state anyway.